Save some memory

pgrawehr opened the pull request:
This saves some RAM by directly reading error messages from Flash memory instead of from RAM. 
Also removed an unused member in the OneWire implementation that reserved 8 bytes * NUMBER_OF_PINS ram for nothing.

pgrawehr commented:
@soundanalogous : Are you still taking PR's on this? I've got some more pending ;-)

pgrawehr commented on src/I2CFirmata.h:
```diff
@@ -108,9 +108,9 @@ void I2CFirmata::readAndReportData(byte address, int theRegister, byte numBytes,
   // check to be sure correct number of bytes were returned by slave
   if (numBytes < Wire.available()) {
-    Firmata.sendString("I2C: Too many bytes received");
+    Firmata.sendString(F("I2C: Too many bytes received"));
```
Bad. Looks like the support for this feature is broken on the AVR, despite it wouldn't even need it (as it has more than enough memory).

pgrawehr replied:
It seems the problem only exists on the MegaAVR boards, such as the Arduino Nano Every. This one isn't supported by this library at all, it seems. When I try to compile for that board, it tells me that the board abstraction in `Board.h` is missing. 

I don't want to just blindly add something there, without testing it (I don't have such a board).

pgrawehr commented on src/OneWireFirmata.cpp:
```diff
@@ -17,6 +17,14 @@
 #include "OneWireFirmata.h"
 #include "Encoder7Bit.h"
 
+OneWireFirmata::OneWireFirmata()
```
Sure, simple thing.

pgrawehr commented:
Closing this, as unfortunatelly there won't be updates in this repo. 
@soundanalogous I'm still interested in taking this over, but yes, that would also mean I'll be making some design decisions.

soundanalogous commented:
If you're committed to work on this, I will add you to the firmata group with full commit access.

pgrawehr commented:
@soundanalogous Sure. As you can see [here](https://github.com/pgrawehr/ConfigurableFirmata/tree/develop), I have already added support for new boards (Arduino Due, ESP32), added some new modules (SPI support, DHT support) and I really intend to continue to do so.

soundanalogous commented:
@pgrawehr do you also intend to maintain [firmata-builder](https://github.com/firmata/firmata-builder) and [firmatabuilder.com](https://github.com/firmata/firmatabuilder.com) as those depend on ConfigurableFirmata?

pgrawehr commented:
@soundanalogous I will try to keep that running, yes. @Ellerbach, who is working with me on the dotnet/iot repo (which includes a modern client implementation for firmata) already said somewhere, that he's willing to take over the main responsibility for the website(s). 

I will create branches at the current state of the repos before adding any changes, so that the current "stable" versions are always available to people who need exactly these versions. I'll update the documentation so the branches can be found. This particularly applies to firmata/protocol, where some of the open proposals imply breaking changes.

Ellerbach commented:
> he's willing to take over the main responsibility for the website(s).

At least the hosting! Then it all depends on the development side what it does require. I4m far to be a (good) front end developer! So I can host the site, help on the pipeline for deployment for sure.

soundanalogous commented:
Ok I'll see if I can find someone with some front end chops to help out as well.

soundanalogous commented:
If you make significant changes to any of the existing features (say you completely redesign how I2C works in Firmata), you could consider using a new ID for that new implementation, that way the legacy prototocol will still be in place for firmata/arduino (StandardFirmata) users.

pgrawehr commented:
Yes, for sure. 

For now, I don't have a reason to break the protocol, except maybe for the issue discussed in https://github.com/firmata/ConfigurableFirmata/issues/100. But that would be a user-configurable setting. The existing components work quite well, so I'm primarily adding functions, not changing existing ones (I haven't looked to much into open issues just yet, though). 

Another thing: Can you please enable branch protection for the master branch (at least on this repo)? This prevents accidental force-pushes to master and so ensures that any change can be reverted. Go to Settings->Branches, click "Add rule" and enter "master" as the branch name. Then click "create" (you can leave all checkboxes unticked).

soundanalogous commented:
Enabled branch protection. Regarding #100, I don't think that is as simple as a user configurable setting. You'd still need to add pin-level (rather than port level as there exists today) reporting. But that can be added as a new feature.
